s3_scrubber/src/lib.rs

```diff
@@ -290,7 +279,7 @@ pub fn init_s3_client(account_id: Option<String>, bucket_region: Region) -> Clie
         );
 
         // Use SSO if we were given an account ID
-        match account_id {
+        match std::env::var("SSO_ACCOUNT_ID").ok() {
```

Comment:
Kind of weird that BucketConfig has `derive(serde::Serialize)`, but that is for two consecutive runs (gather, delete&pray) so perhaps we do not need to worry about backwards compat.. And this change is backwards compat :) Alternatively we could add a `#[serde(deny_unknown_fields)]` to BucketConfig just to be sure someone re-running an older serialization will get an error.

Reply:
added the unknown fields deny.